### ISSUE TITLE
react/prop-types false positive on non-component class

### ISSUE THREAD

**Issue by lencioni** [yannickcr/eslint-plugin-react#45]
If I have a regular class (non-component) that has a property called props, the react/prop-types rule will complain about missing props validation.

```js
class MyClass {
  myMethod() {
    return this.props.test;
  }
}
```

> 3:11  error  'test' is missing in props validation for MyClass  react/prop-types

This should only happen for classes that extend `React.Component` or extend a class that extends React.Component.

eslint-plugin-react 2.0.1

**Comment by yannickcr** [comment]
Right, there is a bug in the React component detection.
